### PR TITLE
refactor: fix component media URLs

### PR DESCRIPTION
**Pull request by JuroOravec**
Further fix for https://github.com/django-components/django-components/issues/1062, as raised in https://github.com/django-components/django-components/pull/1065.

@DNIIBOY Could you test this PR locally to see if it works for your case? IMO it should work, but better to double check.

And sorry for making yet another PR, I don't want to modify other people's PRs (unless it's tiny things) to avoid confusion.

Supersedes https://github.com/django-components/django-components/pull/1065.

### CODE REVIEW

**Comment by DNIIBOY** [comment]
This works in my case, and is a better solution than #1065

File: src/django_components/component_media.py [hunk 1]
```diff
@@ -868,7 +868,11 @@ def resolve_media_file(
     # If the path is a URL, don't resolve it
     # (e.g. https://cdnjs.cloudflare.com/ajax/libs/Chart.js/3.0.2/chart.min.js)
     # This is defined based on Django's `Media.absolute_path()` method.
-    is_url_path = filepath.startswith(("http://", "https", "/"))
+    is_url_path = filepath.startswith(("http://", "https://", "://", "/"))
+
+    # If the path is a URL, don't resolve it
+    if is_url_path:
```
Comment (DNIIBOY) [inline src/django_components/component_media.py:874]:
```python
if allow_glob and is_glob(filepath_abs_or_glob) and not is_url_path:
```
We can remove the `and not is_url_path` from the following if statement for clarity

File: tests/test_component_media.py [hunk 2]
```diff
@@ -415,41 +415,19 @@ def test_glob_pattern_relative_to_root_dir(self):
         assertInHTML('<script src="glob/glob_2.js"></script>', rendered)
 
     def test_glob_pattern_does_not_break_urls(self):
-        class MyComponent(Component):
```
Comment (JuroOravec) [inline tests/test_component_media.py:418]: This is why the edge case slipped the test - When a component is defined within a test file, the test file is not within the directories in `COMPONENTS.dirs`, and so the file paths are not even attempted to be resolved.

So after I moved the component class to be defined in a separate file, which is within `COMPONENTS.dirs`, then that triggered to try to resolve the file paths relative to the component file, which revealed the bug.

File: src/django_components/component_media.py (hunk 1 above)
Comment (JuroOravec) [reply src/django_components/component_media.py:874]: Good catch, thanks!

**Comment by JuroOravec** [comment]
cc @EmilStenstrom

**Comment by JuroOravec** [comment]
@DNIIBOY Thanks again for raising this and your help! This is now released in [0.134](https://github.com/django-components/django-components/releases/tag/0.134)
